### PR TITLE
upgrade to 0.20

### PR DESCRIPTION
**Pull request by 0xmtf**

### CODE REVIEW

File: aggregator_mock/src/lib.rs [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
     pub decimals: u8,
 }
 
-pub type AggregatorResultAsMultiResult<BigUint> =
-    MultiResult5<u32, BoxedBytes, BoxedBytes, BigUint, u8>;
+pub type AggregatorResultAsMultiResult<M> =
+    MultiResult5<u32, ManagedBuffer<M>, ManagedBuffer<M>, BigUint<M>, u8>;
 
 const DEFAULT_PRICE: u64 = 1_000u64;
 const DEFAULT_PRICE_DECIMALS: u8 = 2u8;
```
Comment (dorin-iancu) [inline aggregator_mock/src/lib.rs:18]: No need to have the `u64` and `u8` at the end, you already wrote the type in front.

File: lending_pool/src/factory.rs [hunk 2]
```diff
@@ -7,39 +7,39 @@ pub trait FactoryModule {
...
-        arg_buffer.push_argument_bytes(u_optimal.to_bytes_be().as_slice());
-        arg_buffer.push_argument_bytes(reserve_factor.to_bytes_be().as_slice());
+        let mut arg_buffer = ManagedArgBuffer::new_empty(self.type_manager());
+        arg_buffer.push_arg(base_asset);
+        arg_buffer.push_arg(lending_pool_address);
+        arg_buffer.push_arg(r_base);
+        arg_buffer.push_arg(r_slope1);
+        arg_buffer.push_arg(r_slope2);
+        arg_buffer.push_arg(u_optimal);
+        arg_buffer.push_arg(reserve_factor);
```
Comment (dorin-iancu) [inline lending_pool/src/factory.rs:28]: Could use a deploy proxy here instead. Example: 

https://github.com/ElrondNetwork/sc-bridge-elrond/blob/20fb1c63150deec5228dac03bfaa3233f58d86a4/multisig/src/setup.rs#L75-L79

https://github.com/ElrondNetwork/sc-bridge-elrond/blob/main/multisig/src/setup.rs#L363

File: lending_pool/src/lib.rs [hunk 3]
```diff
@@ -122,15 +116,11 @@ pub trait LendingPool:
...
-            &asset_address,
-            &transfers,
-            &b"repay"[..].into(),
-            &[initial_caller.as_bytes().into()],
-        );
+        let transfers = self.raw_vm_api().get_all_esdt_transfers();
+        self.liquidity_pool_proxy(asset_address)
+            .repay(initial_caller)
+            .with_multi_token_transfer(transfers)
+            .execute_on_dest_context();
```
Comment (dorin-iancu) [inline lending_pool/src/lib.rs:123]: :+1:

File: lending_pool/src/router.rs [hunk 4]
```diff
@@ -55,15 +55,19 @@ pub trait RouterModule: proxy::ProxyModule + factory::FactoryModule {
...
             self.pools_map().contains_key(&base_asset),
             "no pool found for this asset"
         );
 
-        let pool_address = self.pools_map().get(&base_asset).unwrap();
-        let success = self.upgrade_pool(&pool_address, &new_bytecode);
+        let pool_address = self
+            .pools_map()
+            .get(&base_asset)
+            .unwrap_or_else(|| self.types().managed_address_zero());
```
Comment (dorin-iancu) [inline lending_pool/src/router.rs:68]: Should be able to use ManagedAddress::zero(), same as in other similar places.

File: liquidity_pool/src/liquidity.rs [hunk 5]
```diff
@@ -183,29 +181,33 @@ pub trait LiquidityModule:
     #[only_owner]
     #[payable("*")]
     #[endpoint]
-    fn repay(&self, initial_caller: Address) -> SCResult<()> {
+    fn repay(&self, initial_caller: ManagedAddress) -> SCResult<()> {
         self.require_non_zero_address(&initial_caller)?;
 
-        let transfers = self.get_all_esdt_transfers();
-
+        let transfers = self
```
Comment (dorin-iancu) [inline liquidity_pool/src/liquidity.rs:187]: Use `self.call_value().all_esdt_transfers()` instead

File: lending_pool/src/factory.rs (hunk 2 above)
Comment (mateigraura) [reply lending_pool/src/factory.rs:28]: will be refactored in another PR

File: lending_pool/src/router.rs (hunk 4 above)
Comment (mateigraura) [reply lending_pool/src/router.rs]: done

**Review by dorin-iancu — APPROVED** [review]

**Pull request merged**